### PR TITLE
RBAC: Hide root level folder creation button if the user doesn't have the right permissions

### PR DESCRIPTION
**Pull request by IevaVasiljeva**
**What is this feature?**

A follow up on https://github.com/grafana/grafana/pull/91215, specifically on https://github.com/grafana/grafana/pull/91215#issuecomment-2262617529 - hides the folder creation button at the root level for users who do not have permissions to create folders there (eg, org Viewers who have edit rights only to selected subfolders, so should only be able to create folders there).

This is how the root dashboard page will look for users without access to create folders there:
Screenshot 2024-08-02 at 17 50 00

**Why do we need this feature?**

To hide functionality that user is not allowed to carry out.

**Who is this feature for?**

Anyone

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/identity-access-team/issues/803

**Special notes for your reviewer:**

I'm not sure this is the best approach, as the code is not super readable. Some alternatives:
* we could also fetch the general folder data somewhere else (not sure where);
* we could simplify the logic in `permissions.ts` by not ignoring the general folder permissions (as in, we could skip [this](https://github.com/grafana/grafana/compare/ieva/hide-folder-creation-button?expand=1#diff-86ed66e30ed1e77f097b6dbe391b7139607841a60035f5ed89f38e89cb2833d9R7-R9)) - this would mean that Viewer users would by default not see the `New` button at the root `/dashboards` page.

### CODE REVIEW

File: public/app/features/browse-dashboards/BrowseDashboardsPage.tsx [hunk 1]
```diff
@@ -87,7 +87,9 @@ const BrowseDashboardsPage = memo(({ match }: Props) => {
 
   const hasSelection = useHasSelection();
 
-  const { canEditFolders, canEditDashboards, canCreateDashboards, canCreateFolders } = getFolderPermissions(folderDTO);
+  const { data: rootFolder } = useGetFolderQuery('general');
```
Comment (eleijonmarck) [inline public/app/features/browse-dashboards/BrowseDashboardsPage.tsx:90]: @IevaVasiljeva maybe i "dumb" question. but will this always be 'general' . This is querying UUID?

```ts
    // get folder info (e.g. title, parents) but *not* children
    getFolder: builder.query<FolderDTO, string>({
      providesTags: (_result, _error, folderUID) => [{ type: 'getFolder', id: folderUID }],
      query: (folderUID) => ({ url: `/folders/${folderUID}`, params: { accesscontrol: true } }),
    }),
```

https://github.com/grafana/grafana/blob/main/public/app/features/browse-dashboards/api/browseDashboardsAPI.ts

**Review by Jguer — APPROVED** [review]

**Comment by eleijonmarck** [comment]
@IevaVasiljeva i tested this and found that there is still the button to create a folder for a viewer in the root directory.

i created `viewer-viewer, viewer-editor, viewer-admin` folders in main branch
i switched to this branch
logged in as a viewer
able to see the create folder button

i am reproducing this wrong?

**Comment by eleijonmarck** [comment]
Previously i posted a false claim that I was able to reproduce the create folder button. It was caching + none-built frontend code.

This fixes the issue of being able to see the "new folder" button for "root" directory.

**Review by eleijonmarck — APPROVED** [review]
LGTM

**Comment by IevaVasiljeva** [comment]
> It was caching + none-built frontend code.

The good old wrong frontend issue 😄 Thanks for testing!

**Pull request merged**